src/include/ipc/dai-imx.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright 2019 NXP
+ *
+ * Author: Daniel Baluta <daniel.baluta@nxp.com>
+ */
+
+#ifndef __IPC_DAI_IMX_H__
+#define __IPC_DAI_IMX_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/* ESAI Configuration Request - SOF_IPC_DAI_ESAI_CONFIG */
+struct sof_ipc_dai_esai_params {
+	struct sof_ipc_hdr hdr;
+
+	/* MCLK */
+	uint16_t reserved1;
+	uint16_t mclk_id;
+	uint32_t mclk_rate; /* MCLK frequency in Hz */
+	uint32_t mclk_direction;
+
+	/* TDM */
+	uint32_t tdm_slots;
```

Comment:
don't you support independent slots numbers on RX and TX with the 6-wire interface?

Reply:
yes, we support independent RX/TX with 6 wire interface. Anyhow, it looks like ASoC API (e.g snd_soc_dai_set_tdm_slot) prefers to pass tdm_slots and a mask of active rx/tx. I want to follow that for now.